src/WorkerPool.js

```diff
@@ -59,14 +67,26 @@ class WorkerPool {
     }
   }
 
+  cancel (runId) {
+    const info = this.runInfo[runId]
+    if (info) {
+      info.canceled = true
+    }
+  }
+
   // todo: change to #addTask(resultIndex, taskArgs) { after private methods
   // proposal accepted and supported by default in Babel.
   addTask (infoIndex, resultIndex, taskArgs) {
     const info = this.runInfo[infoIndex]
+
+    if (info && info.canceled) {
+      info.reject('Remaining tasks canceled')
```

Comment:
*cancelled*

Reply:
We could also add cancel checks around line 74 and line 115.

Reply:
> We could also add cancel checks around line 74 and line 115.

Yeah that makes sense, I'll add them there as well.